src/frontend/Komponenter/Behandling/VedtakOgBeregning/Overgangsstønad/InnvilgeVedtak/InnvilgeVedtak.tsx

```diff
@@ -96,12 +114,47 @@ export const InnvilgeVedtak: React.FC<{
     const periodeBegrunnelse = formState.getProps('periodeBegrunnelse') as FieldState;
     const inntektBegrunnelse = formState.getProps('inntektBegrunnelse') as FieldState;
     const typeSamordningsfradag = formState.getProps('samordningsfradragType') as FieldState;
+    const yngsteBarnDato = formState.getProps('yngsteBarnDato') as FieldState;
 
     const inntektsperioder = inntektsperiodeState.value;
     const vedtaksperioder = vedtaksperiodeState.value;
 
     const låsVedtaksperiodeRad = !!revurderesFra;
 
+    const yngsteBarnFødselsdato = useCallback(
+        (vilkår: RessursSuksess<IVilkår>) => {
+            const terminbarnFødselsdatoer = vilkår.data.grunnlag.barnMedSamvær
+                .map((b) => b.søknadsgrunnlag)
+                .map((sb) => sb.fødselTermindato)
+                .filter(
+                    (fødselTermindato): fødselTermindato is string =>
+                        !!fødselTermindato &&
+                        erGyldigDato(fødselTermindato) &&
+                        erEtter(fødselTermindato, minusÅr(new Date(), 1))
+                );
+
+            const tidligsteDato = vilkår.data.grunnlag.barnMedSamvær
+                .map((b) => b.registergrunnlag)
+                .map((r) => r.fødselsdato)
+                .filter(
+                    (fødselsdato): fødselsdato is string =>
+                        !!fødselsdato && erGyldigDato(fødselsdato)
+                )
+                .concat(terminbarnFødselsdatoer)
+                .reduce((a, b) => {
+                    return erEtter(a, b) ? a : b;
+                });
```

Comment:
Kanskje vi kan flytte ut selve "finnTidligsteDato" til en metode utenfor selve komponenten, eller egen fil? 
```suggestion
            const barnMedSamvær = vilkår.data.grunnlag.barnMedSamvær;
            const tidligsteDato = [
                ...barnMedSamvær.map((b) => b.søknadsgrunnlag.fødselTermindato),
                ...barnMedSamvær.map((b) => b.registergrunnlag.fødselsdato),
            ]
                .filter(
                    (fødselTermindato): fødselTermindato is string =>
                        !!fødselTermindato && erGyldigDato(fødselTermindato)
                )
                .reduce((a, b) => (erEtter(a, b) ? a : b));
```

Reply:
Trenger vel kanskje `&& erEtter(fødselTermindato, minusÅr(new Date(), 1))` også

Reply:
Vi tar ikke med "gamle" termindatoer (over ett år gamle) i vurderingen derfor "erEtter" kun på søknadsgrunnlaget. Registerbarn sjekker vi alle. Her kan det i teorien bli surr mellom registerbarn og søknadsbarn, men siden vi skal bruke denne for å sjekke alder er under 8 ved utgang av innvilgelse av hovedperiode tenker vi dette er "godt nok". 

Så hvis du har barn som er 7,8 og 9 tar vi med disse. Dersom de eneste barna er 8 og 9 vil validering gi feil. Dersom barnet er 7  vil det kanskje gi feil (avh av tom dato på periode) 

Dersom det finnes barn på 8 og 9, men en termindato i søknadsdata => alt ok dersom termindato ikke er "gammel".  

Tror vi kan havne i teoretisk "falsk negativ" (skulle ikke blitt regnet med når vi validerer) dersom det finnes ikke eksisterende terminbarn i søknadsdata, men igjen - tenker kanskje det er godt nok.

Reply:
Ok 👍